collectors/ciscomeraki/collector.js

```diff
@@ -0,0 +1,270 @@
+            let differenceNetworks = utils.differenceOfNetworksArray(networks, networksFromS3);
+            AlLogger.debug(`CMRI0000024 Networks updated ${JSON.stringify(differenceNetworks)}`);
+
+            if (differenceNetworks.length > 0) {
+                const startTs = process.env.paws_collection_start_ts ?
+                    process.env.paws_collection_start_ts :
+                    moment().toISOString();
+                const endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+                const initialStates = differenceNetworks.map(networkId => ({
+                    networkId: networkId,
+                    since: startTs,
+                    until: endTs,
+                    nextPage: null,
+                    poll_interval_sec: parseInt(Math.floor(Math.random() * 30) + 1)
+                }));
+                AlLogger.debug(`CMRI0000020: SQS message added ${JSON.stringify(initialStates)}`);
+                collector._storeCollectionState({}, initialStates, this.pollInterval, () => {
+                    utils.uploadNetworksListInS3Bucket(keyValue, networks);
+                });
+            } else {
+                utils.uploadNetworksListInS3Bucket(keyValue, networks);
+            }
+        } else if (networksFromS3 && (networksFromS3.Code === 'NoSuchKey' || networksFromS3.Code === 'AccessDenied')) {
+            await utils.uploadNetworksListInS3Bucket(keyValue, networks);
+        }
+    }
+
+    pawsGetLogs(state, callback) {
+        const collector = this;
+
+        const productTypes = process.env.paws_collector_param_string_1 ? JSON.parse(process.env.paws_collector_param_string_1) : [];
+        if (!productTypes) {
+            return callback("The Product Types was not found!");
+        }
+        const { clientSecret, apiEndpoint, orgKey } = this.validateAndPrepare(callback);
+        const apiDetails = utils.getAPIDetails(orgKey, productTypes);
+        if (!apiDetails.url) {
+            return callback("The API name was not found!");
+        }
+        AlLogger.info(`CMRI000001 Collecting data for NetworkId-${state.networkId} from ${state.since}`);
+        utils.getAPILogs(apiDetails, [], apiEndpoint, state, clientSecret, process.env.paws_max_pages_per_invocation)
+            .then(({ accumulator, nextPage }) => {
+                let newState;
+                if (nextPage === undefined) {
+                    newState = this._getNextCollectionState(state);
+                } else {
+                    newState = this._getNextCollectionStateWithNextPage(state, nextPage);
+                }
+                AlLogger.info(`CMRI000002 Next collection in ${newState.poll_interval_sec} seconds`);
+                return callback(null, accumulator, newState, newState.poll_interval_sec);
+            })
+            .catch((error) => {
+                if (error && error.response && error.response.status == 429) {
```

Comment:
can we  add this code in separate function.
handleThrottlingError(error, state, callback) and handleOtherErrors(error, state, callback).